content/blog/2022/10/2022-10-10-plugin-health-scoring-system-report.adoc

```diff
@@ -0,0 +1,130 @@
+
+---
+
+== Overview
+
+. <<About Project>>
+. <<Architecture Diagram>>
+. <<Phase 1>>
+. <<Phase 2>>
+. <<Next Steps>>
+. <<Acknowledgments>>
+. <<Useful Links>>
+
+---
+
+== About Project
+
+Plugin maintenance is an extremely important phase of a plugin’s lifecycle. Jenkins has always been active in ongoing developments which raises a need for thousands of plugins to catch up with their latest developments, to stay useful for a broader set of users.
+
+We aim to fulfill this need by introducing a metric system to calculate the health score of Jenkins plugins. This health score measures the plugin’s development maturity level with the help of accurate numeric figures. This score attempts to provide an accurate picture of how much care and help a plugin needs and makes it easier for contributors to contribute. This score also allows users to make a conscious decision before installing and/or using a plugin.
+
+There were two coding phases in the GSoC 2022.
+
+---
+
+== Architecture Diagram
+
+image::/images/gsoc/2022/plugin-health-score/architecture-diagram.png[Architecture Diagram, height=690, role=center, float=center]
+
+---
+
+== Phase 1
+
+In this phase, we worked on setting up the project and followed best practices while doing so. The whole application is dockerized so that it is easier for anyone to spin up this project and test it out locally. There's continuous integration (CI) in place for the project to make sure that new changes are tested automatically. The useful details from the Jenkins Update Center are stored inside the database to run analysis.
+
+=== Deliverables
+
+* Create a spring boot project connected with a PostgreSQL DB (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/3[PR])
+* Dockerize the whole application (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/7[PR])
+* Setup continuous integration for the project (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/10[PR])
+* Store plugin details from the update center, inside the DB in JSON format (link:https://github.com/jenkins-infra/plugin-health-scoring/pull/18[PR])
+
+=== Resources
+
+* link:https://github.com/jenkins-infra/plugin-health-scoring/milestone/1?closed=1[View Milestone 1]
+* Jenkins Online Meetup Phase 1: link:https://docs.google.com/presentation/d/1t2vuNn1NFpDusnw0m4vdFw6WBQMeU6kccv_K1v2L6R0/edit#slide=id.g13dcaed2105_0_0[Slides]
+* Jenkins Online Meetup Phase 1: Demo and Presentation
+
+video::loLSNdCv6K4[youtube,width=800,height=420,start=1089]
+
+---
+
+== Phase 2
+
+In this phase, the probe engine for the probes was designed. And processes like reading the update center and executing the probe engine are now running on a schedule without the need to restart the application. There are additions of useful probes running in the probe engine, which are getting listed on the UI to highlight the current state of the project.
```

Comment:
```suggestion
In this phase, the probe engine for the probes was designed.
Processes like reading the update center and executing the probe engine are now running on a schedule, without the need to restart the application.
There are additional useful probes running in the probe engine, which are listed on the UI to highlight the current state of the project.
```
small wording suggestion, if I am misinterpreting the idea of the message, please let me know
otherwise, let me know if those make sense!

Reply:
Looks great, thanks for your suggestion!

Regarding,

> There are additional useful probes running in the probe engine...

As per my understanding, this sentence gives a sense that some additional probes were added on top of some already-existing probes. But that's not true, as there were no existing probes before these ones.

What do you think?

Reply:
Thanks for sharing your insight!

In that case,  would "The creation of probes" or "Probes are created at this time" be an acceptable alternative instead? If there are no pre-existing probes, making that clear would be helpful in this case. However, if that doesn't make sense in terms of the actual process/function, then disregard my suggestion.

Reply:
Awesome!

Let's go with "Probes created at this time are getting run by the probe engine, and are listed on the UI to highlight the current state of the project." (Please correct if there are any grammatical errors)

Since this project was created from scratch and there was no mention of "probes" in Phase 1, I think we can go on without explicitly mentioning that there were no pre-existing probes before this.